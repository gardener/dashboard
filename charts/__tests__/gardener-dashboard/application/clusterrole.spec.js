//
// SPDX-FileCopyrightText: 2021 SAP SE or an SAP affiliate company and Gardener contributors
//
// SPDX-License-Identifier: Apache-2.0
//

'use strict'

const { basename } = require('path')
const { helm } = fixtures

const renderTemplates = helm.renderTemplatesFn('gardener-dashboard', 'charts', basename(__dirname))

describe('gardener-dashboard', function () {
  describe('clusterrole', function () {
    let templates

    beforeEach(() => {
      templates = [
        'clusterrole'
      ]
    })

    it('should render the template with default values', async function () {
      const values = {}
      const documents = await renderTemplates(templates, values)
      expect(documents).toHaveLength(1)
      const [clusterRole] = documents
      expect(clusterRole).toMatchSnapshot()
    })
  })
})
