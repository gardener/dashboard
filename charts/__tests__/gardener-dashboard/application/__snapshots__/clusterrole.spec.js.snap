// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`gardener-dashboard clusterrole should render the template with default values 1`] = `
Object {
  "apiVersion": "rbac.authorization.k8s.io/v1",
  "kind": "ClusterRole",
  "metadata": Object {
    "labels": Object {
      "app": "gardener-dashboard",
      "chart": "application-0.1.0",
      "heritage": "Helm",
      "release": "application",
    },
    "name": "gardener.cloud:system:dashboard",
  },
  "rules": Array [
    Object {
      "apiGroups": Array [
        "authentication.k8s.io",
      ],
      "resources": Array [
        "tokenreviews",
      ],
      "verbs": Array [
        "create",
      ],
    },
    Object {
      "apiGroups": Array [
        "core.gardener.cloud",
      ],
      "resources": Array [
        "quotas",
      ],
      "verbs": Array [
        "list",
        "watch",
      ],
    },
    Object {
      "apiGroups": Array [
        "core.gardener.cloud",
      ],
      "resources": Array [
        "projects",
      ],
      "verbs": Array [
        "list",
        "watch",
      ],
    },
    Object {
      "apiGroups": Array [
        "core.gardener.cloud",
      ],
      "resources": Array [
        "shoots",
      ],
      "verbs": Array [
        "list",
        "watch",
      ],
    },
    Object {
      "apiGroups": Array [
        "apiregistration.k8s.io",
      ],
      "resources": Array [
        "apiservices",
      ],
      "verbs": Array [
        "get",
      ],
    },
    Object {
      "apiGroups": Array [
        "core.gardener.cloud",
      ],
      "resources": Array [
        "controllerregistrations",
      ],
      "verbs": Array [
        "list",
        "watch",
      ],
    },
    Object {
      "apiGroups": Array [
        "",
      ],
      "resourceNames": Array [
        "cluster-identity",
      ],
      "resources": Array [
        "configmaps",
      ],
      "verbs": Array [
        "get",
      ],
    },
    Object {
      "apiGroups": Array [
        "",
      ],
      "resources": Array [
        "secrets",
      ],
      "verbs": Array [
        "get",
      ],
    },
    Object {
      "apiGroups": Array [
        "core.gardener.cloud",
      ],
      "resources": Array [
        "shoots",
      ],
      "verbs": Array [
        "get",
      ],
    },
  ],
}
`;
